### PR TITLE
OIDC Federation Test Case

### PR DESCRIPTION
**Pull request by Chamodii**

### CODE REVIEW

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oidc/OIDCIdentityFederationTestCase.java [hunk 1]
```diff
@@ -0,0 +1,648 @@
...
+     *
+     * @param client
+     * @param urlParameters
+     * @param url
+     * @return
+     * @throws ClientProtocolException
+     * @throws java.io.IOException
+     */
+    public HttpResponse sendPostRequestWithParameters(HttpClient client, List<NameValuePair> urlParameters, String url)
+            throws ClientProtocolException,
```
Comment (piraveena) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oidc/OIDCIdentityFederationTestCase.java]: Add new line

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oidc/OIDCIdentityFederationTestCase.java [hunk 2]
```diff
@@ -0,0 +1,648 @@
...
+
+    private String usrName = "testFederatedUser";
+    private String usrPwd = "testFederatePassword";
+    private String usrRole = "admin";
+
+    protected OauthAdminClient adminClient;
+
+    CookieStore cookieStore = new BasicCookieStore();
+
+
```
Comment (piraveena) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oidc/OIDCIdentityFederationTestCase.java]: Remove extra line

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oidc/OIDCIdentityFederationTestCase.java [hunk 3]
```diff
@@ -0,0 +1,648 @@
...
+                }
+            }
+        }
+
+        Assert.assertTrue(success, "Failed to update service provider with inbound SAML2 configs in primary IS");
+        Assert.assertTrue(AUTHENTICATION_TYPE.equals(serviceProvider.getLocalAndOutBoundAuthenticationConfig().
+                getAuthenticationType()), "Failed to update local and out bound configs in primary IS");
+    }
+
+
```
Comment (piraveena) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oidc/OIDCIdentityFederationTestCase.java]: Remove extra line

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oidc/OIDCIdentityFederationTestCase.java [hunk 4]
```diff
@@ -0,0 +1,648 @@
...
+     *
+     * @param client      - http Client
+     * @param locationURL - Get url location
+     * @return http response
+     * @throws ClientProtocolException
+     * @throws java.io.IOException
+     */
+    public HttpResponse sendGetRequest(HttpClient client, String locationURL)
+            throws
+            ClientProtocolException,
```
Comment (piraveena) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oidc/OIDCIdentityFederationTestCase.java]: Add new line

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oidc/OIDCIdentityFederationTestCase.java [hunk 5]
```diff
@@ -0,0 +1,648 @@
...
+        }
+
+        HttpResponse response = sendPostRequestWithParameters(client, urlParameters, "https://localhost:9854/oauth2" +
+                "/authorize");
+        return response;
+    }
+
+    /**
+     * Send Post request
+     * testfederationtestfederation
```
Comment (piraveena) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oidc/OIDCIdentityFederationTestCase.java]: Comment is not meaningful

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oidc/OIDCIdentityFederationTestCase.java [hunk 6]
```diff
@@ -0,0 +1,648 @@
...
+     * @param client      - http Client
+     * @param locationURL - Post url location
+     * @return http response
+     * @throws ClientProtocolException
+     * @throws java.io.IOException
+     */
+    public HttpResponse sendPostRequest(HttpClient client, String locationURL)
+            throws ClientProtocolException,
+            IOException {
+        HttpPost postRequest = new HttpPost(locationURL);
```
Comment (piraveena) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oidc/OIDCIdentityFederationTestCase.java]: foratting-adfd new line

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oidc/OIDCIdentityFederationTestCase.java [hunk 7]
```diff
@@ -0,0 +1,698 @@
...
+
+        return locationHeader;
+    }
+
+    @Test(priority = 4, groups = "wso2.is", description = "Check login flow of primary IS service provider")
+    public void testFederation() throws Exception {
+
+        HttpClient client = HttpClientBuilder.create().setDefaultCookieStore(cookieStore).build();
+
+        String sessionId = sendSAMLRequestToPrimaryIS(client);
```
Comment (darshanasbg) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oidc/OIDCIdentityFederationTestCase.java]: test 1

**Comment by darshanasbg** [comment]
Test that includes logout flow also has been sent with the PR: https://github.com/wso2/product-is/pull/7039. Hence closing this.

**Pull request closed without merging**